src/context_menu/index.jsx

```diff
@@ -29,10 +29,19 @@ class ContextMenu extends PureComponent {
   }
 
   activate() {
-    if (this.isListenerActive) return;
-    this.stopListeningToKeys = keymanager('esc', this.hide);
-    document.addEventListener('click', this.handleGlobalClick);
-    this.isListenerActive = true;
+    // # Workaround for event bubbling issue in react 17
+    // Prevents the following:
+    // - Some click handler calls ContextMenu.show()
+    // - ContextMenu attaches global click listener
+    // - Click event bubbles up from react root node to document
+    // - ContextMenu global click handler is called
+    setTimeout(() => {
+      if (this.isListenerActive) return;
```

Comment:
need to check that component is mounted. Otherwise will produce memory leak.

Another way to not use timeout is to pass activate to setState as callback on line
https://github.com/goodhood-eu/nebenan-components/blob/9ae99e0ef5bb18fccb66d415fe1e582f8cfa8f9f/src/context_menu/index.jsx#L61
this way `activate` will be called only after state update and after global click.